### DISCUSSION TITLE
how to go through the repo

### DISCUSSION THREAD

**Discussion by itsinthebag** [prestodb/presto#15228] [General] [answered]
I'm really want to learn presto, but I don't know where to start, which piece of code should I understand first. 
Please help!

**Comment by bitsondatadev** [comment, marked as answer]
Hey @itsinthebag! The best resource to date IMO to get you started is [The Presto Definitive Guide](https://trino.io/presto-the-definitive-guide.html) that was written by one of the Presto co-creators. Aside, the co-creators now work on a Presto fork called [Trino](https://trino.io).

If you are interested in jumping into the code, it's easiest to start with one of the connectors where you are familiar with the model already. For example, [I started contributing in the Elasticsearch connector](https://trino.io/blog/2020/10/01/intro-developer-advocate.html) because I was most familiar with that data model and the client code. 

Working with the connector code will slowly introduce you to more core areas of the project and you can try to do other things like implement your own connector by mimicking [the example connector](https://trino.io/docs/current/develop/example-http.html), or update some part of the [query planner](https://trino.io/docs/current/optimizer/cost-based-optimizations.html). We cover a lot of these concepts in the [community broadcast](https://trino.io/broadcast/) we do every 2 weeks.

If you have any questions feel free to reach out to me on the [Trino slack channel](https://trino.io/slack.html).

**Reply by itsinthebag** [reply]
thanks a lot for the reply
